src/fiatconnect/FiatDetailsScreen.tsx

```diff
@@ -0,0 +1,273 @@
+        body[formFields[i].name] = inputRefs.current[i]
+      }
+
+      // TODO: some schema requires additional fields in body in addition
+      // to the ones collected from the form, eg. country code
+      await addNewFiatAccount(providerURL, fiatAccountSchema, body)
+        .then((data) => {
+          // TODO Tracking here
+          dispatch(showMessage(t('fiatDetailsScreen.addFiatAccountSuccess')))
+        })
+        .catch((error) => {
+          // TODO Tracking here
+          if (error === FiatConnectError.ResourceExists) {
+            dispatch(showError(ErrorMessages.ADD_FIAT_ACCOUNT_RESOURCE_EXIST))
+          }
+          Logger.error(TAG, `Error adding fiat account: ${error}`)
+        })
+
+      // TODO: navigate to the next screen
+    }
+  }
+
+  const onPressSelectedPaymentOption = () => {
+    // TODO: tracking here
+    // todo: naviagte to select payment options screen
+  }
+
+  const validateInput = () => {
+    setValidInputs(false)
+    const newErrorSet = new Set<string>()
+
+    let hasEmptyFields = false
+    formFields.forEach((field, index) => {
+      const fieldVal = inputRefs.current[index].trim()
+
+      if (!fieldVal) {
+        hasEmptyFields = true
+      } else if (!field.regex.test(fieldVal)) {
+        newErrorSet.add(field.name)
+      }
+    })
+
+    setErrors(newErrorSet)
+    setValidInputs(!hasEmptyFields && newErrorSet.size === 0)
+  }
+
+  const setInputValue = (value: string, index: number) => {
+    inputRefs.current[index] = value
+    setTextValue(value)
+
+    validateInput()
+  }
+
+  return (
+    <SafeAreaView style={styles.container}>
+      <ScrollView>
+        {formFields.map((field, index) => {
+          return (
+            <View style={styles.inputView} key={`inputField-${index}`}>
+              <Text style={styles.inputLabel}>{field.label}</Text>
+              <TextInput
+                testID={`input-${field.name}`}
+                style={styles.formInput}
+                value={inputRefs.current[index]}
+                placeholder={field.placeholderText}
+                onChangeText={(value) => setInputValue(value, index)}
+              />
+              {errors.has(field.name) && (
+                <Text testID="errorMessage" style={styles.error}>
+                  {field.errorMessage}
+                </Text>
+              )}
+            </View>
+          )
+        })}
+      </ScrollView>
+
+      <View style={styles.footer}>
+        <View style={styles.paymentOption}>
+          <BorderlessButton onPress={onPressSelectedPaymentOption} testID="selectedProviderButton">
+            <View style={styles.paymentOptionButton}>
+              <Text style={styles.paymentOptionText}>
+                {t('fiatDetailsScreen.selectedPaymentOption')}
+              </Text>
+              <ForwardChevron color={colors.gray4} />
+              {/* TODO: This is a hardcoded logo, replace the logo uri with the uri from the payment provider selected */}
```

Comment:
Can you add the image url into props? That might not be where it lives in the end, but it helps ensure it won't be forgotten.